### PR TITLE
Implements 'file' method (Issue#144)

### PR DESCRIPTION
**Pull request by bin-doph**
Adds 'file' to serve a single file + tests for both API's, enjoy

### CODE REVIEW

**Comment by Emeric54** [comment]
You should create a new branch to make your changes. That would for example avoid the first 6 commits, which are useless. 🦊

**Comment by szabgab** [comment]
@bin-doph thanks for your PR. If you need git-help with what @Emeric54 suggested, talk to him on the Slack channel.

**Comment by bin-doph** [comment]
Sorry, my bad, won't happen again. Still getting used to... things and stuff. :octocat:

File: t/18-serve-single-file.t [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+my $route-fail = "/fail";
+my $method = "GET";
+
+config.file-discovery-content-type = $mime-type;
+
+# Setup routes
+get $route-default      => sub { render-file("data/file.txt"); }
+get $route-explicit     => sub { render-file("data/file.txt", :$mime-type); }
+get $route-unknown      => sub { render-file("data/unknown.extention"); } # should be extention :(
+#get $route-    => sub { render-file("data/unknown.extension", :$mime-type); }
```
Comment (ufobat) [inline t/18-serve-single-file.t]: could you please complete the test case and then i am going to merge it in.

**Comment by ufobat** [comment]
it's in, thanks :+1:

**Pull request closed without merging**